From 5e4ff519711e10fb938387bd87c4f262ca3c1435 Mon Sep 17 00:00:00 2001
From: Corey Bryant <corey.bryant@canonical.com>
Date: Tue, 6 Jan 2015 15:41:29 -0500
Subject: [PATCH] Handle ISCSIConnector.get_initiator() OSError

Handle ISCSIConnector.get_initiator() OSError exception (errno=ENOENT)
when /etc/iscsi/initiatorname.iscsi doesn't exist.

Closes-Bug: 1408087
---
 cinder/brick/initiator/connector.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cinder/brick/initiator/connector.py b/cinder/brick/initiator/connector.py
index 56d6c91..e6abab1 100644
--- a/cinder/brick/initiator/connector.py
+++ b/cinder/brick/initiator/connector.py
@@ -13,6 +13,7 @@
 #    License for the specific language governing permissions and limitations
 #    under the License.
 
+import errno
 import os
 import socket
 import time
@@ -322,7 +323,9 @@ class ISCSIConnector(InitiatorConnector):
             for l in lines.split('\n'):
                 if l.startswith('InitiatorName='):
                     return l[l.index('=') + 1:].strip()
-        except putils.ProcessExecutionError:
+        except (putils.ProcessExecutionError, OSError) as err:
+            if isinstance(err, OSError) and err.errno != errno.ENOENT:
+                raise
             msg = (_("Could not find the iSCSI Initiator File %s")
                    % file_path)
             LOG.warn(msg)
-- 
2.1.0

